core/notification/src/main/java/edu/stanford/spezi/core/notification/setting/NotificationSettings.kt

```diff
@@ -0,0 +1,43 @@
+package edu.stanford.spezi.core.notification.setting
+
+import edu.stanford.spezi.core.design.action.PendingActions
+
+internal enum class NotificationType(
+    val key: String,
+    val section: Section,
+) {
+    APPOINTMENT_REMINDERS(key = "receivesAppointmentReminders", section = Section.REMINDERS),
+    MEDICATION_UPDATES(key = "receivesMedicationUpdates", section = Section.UPDATES),
+    QUESTIONNAIRE_REMINDERS(key = "receivesQuestionnaireReminders", section = Section.REMINDERS),
+    RECOMMENDATION_UPDATES(key = "receivesRecommendationUpdates", section = Section.UPDATES),
+    VITALS_REMINDERS(key = "receivesVitalsReminders", section = Section.REMINDERS),
+    WEIGHT_ALERTS(key = "receivesWeightAlerts", section = Section.TRENDS),
+    ;
+
+    enum class Section {
+        REMINDERS, UPDATES, TRENDS
+    }
+}
+
+internal data class NotificationSettings(
+    private val settings: Map<NotificationType, Boolean>,
+    val pendingActions: PendingActions<NotificationType> = PendingActions(),
```

Comment:
I would move this in the `NotificationSettingsLoaded` type instead and use `SwitchChanged` as the action type